mono/metadata/threadpool-worker-default.c

```diff
@@ -123,10 +123,10 @@ typedef union {
 typedef struct {
 	MonoRefCount ref;
 
-	MonoThreadPoolWorkerCallback callback;
-
 	ThreadPoolWorkerCounter counters;
 
+	MonoThreadPoolWorkerCallback callback;
+
```

Comment:
This is only going to work by accident.  It might be better to use the  __aligned__ attribute:
https://gcc.gnu.org/onlinedocs/gcc-3.3/gcc/Type-Attributes.html